src/test/java/com/google/api/generator/engine/ast/TryCatchStatementTest.java

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import java.util.Arrays;
+import org.junit.Test;
+
+public class TryCatchStatementTest {
+
+  @Test
+  public void validTryCatchStatement_simple() {
+    Reference exceptionReference = Reference.withClazz(IllegalArgumentException.class);
+    TypeNode type = TypeNode.withReference(exceptionReference);
+    VariableExpr variableExpr =
+        VariableExpr.builder().setVariable(createVariable("e", type)).setIsDecl(true).build();
+
+    TryCatchStatement tryCatch =
+        TryCatchStatement.builder()
+            .setTryBody(Arrays.asList(ExprStatement.withExpr(createAssignmentExpr())))
+            .setCatchVariableExpr(variableExpr)
+            .build();
+    assertThat(tryCatch.catchVariableExpr()).isEqualTo(variableExpr);
+  }
+
+  @Test
+  public void validTryCatchStatement_withResources() {
+    Reference exceptionReference = Reference.withClazz(IllegalArgumentException.class);
+    TypeNode type = TypeNode.withReference(exceptionReference);
+    VariableExpr variableExpr =
+        VariableExpr.builder().setVariable(createVariable("e", type)).setIsDecl(true).build();
+    AssignmentExpr assignmentExpr = createAssignmentExpr();
+
+    TryCatchStatement tryCatch =
+        TryCatchStatement.builder()
+            .setTryResourceExpr(assignmentExpr)
+            .setTryBody(Arrays.asList(ExprStatement.withExpr(assignmentExpr)))
+            .setCatchVariableExpr(variableExpr)
+            .build();
+    assertThat(tryCatch.catchVariableExpr()).isEqualTo(variableExpr);
+    assertThat(tryCatch.tryResourceExpr()).isEqualTo(assignmentExpr);
+  }
+
+  @Test
+  public void validTryCatchStatement_sampleCode() {
+    Reference exceptionReference = Reference.withClazz(IllegalArgumentException.class);
+    TypeNode type = TypeNode.withReference(exceptionReference);
+    VariableExpr variableExpr =
+        VariableExpr.builder().setVariable(createVariable("e", type)).setIsDecl(true).build();
+
+    TryCatchStatement tryCatch =
+        TryCatchStatement.builder()
+            .setTryBody(Arrays.asList(ExprStatement.withExpr(createAssignmentExpr())))
+            .setIsSampleCode(true)
+            .build();
```

Comment:
Here `isSampleCode` set to true to cover the case [here](https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline#L2668) that we only have try block without catch block, right?

Reply:
Yup.

Reply:
👍 Thanks, LGTM!